nixos/modules/system/boot/systemd.nix

```diff
@@ -606,6 +615,16 @@ in
     systemd.targets.network-online.wantedBy = [ "multi-user.target" ];
     systemd.services.systemd-importd.environment = proxy_env;
     systemd.services.systemd-pstore.wantedBy = [ "sysinit.target" ]; # see #81138
+    systemd.services.systemd-networkd-wait-online.serviceConfig.ExecStart = mkIf
```

Comment:
I guess it's better to use `systemd.network.wait-online.extraArgs` to make sure that other arguments configured aren't silently discarded.